### PR TITLE
Wishlist cart functionality

### PR DESCRIPTION
**Pull request by abhishektanwar**
Wishlist features :
- Add / Remove product to wishlist from product card
- Move product to wishlist from cart

Cart features :
- Add product to cart from product card
- Move product to cart from wishlist
- Increase/Decrease quantity and Delete product from the cart

Files to check :
- src/Contexts/CartContext/index.jsx
- src/Contexts/WishlistContext/index.jsx
- src/Hooks/useProductCard.js
- src/components/ProductCard/index.jsx

### CODE REVIEW

File: src/components/ProductCard/index.jsx [hunk 1]
```diff
@@ -1,8 +1,169 @@
...
+import utils from "../../utils";
 import Button from "../Header/Button";
+import { toast } from "react-toastify";
 import "./product-card.css";
-// imagePath={imageUrl} badgeText={'New Arrival'} title = {'One ear 100 headphones black'} price={1890} actualPrice={2099} discountPercentage={10} 
-const ProductCard = ({variant,product,cartActionBtnContainer}) => {
+import { useAuth } from "../../Contexts/AuthDialogContext";
+import { useModal } from "../../Contexts/ModalContext";
+import { useLocation } from "react-router-dom";
+import { useCart } from "../../Contexts/CartContext";
```
Comment (Shubhamkr2610) [inline src/components/ProductCard/index.jsx:7]: Suggestion: you can use absolute import , way of importing file looks more clean and understandable .
reference: you can read about it here https://javascript.plainenglish.io/why-and-how-to-use-absolute-imports-in-react-d5b52f24d53c

File: src/components/ProductCard/index.jsx [hunk 2]
```diff
@@ -1,8 +1,169 @@
...
+  //     { action: { type } },
+  //     { headers: { authorization: token } }
+  //   );
+  // };
+
+  // export const removeProductCartService = async (productId, token) => {
+  //   return await axios.delete(`/api/user/cart/${productId}`, {
+  //     headers: { authorization: token },
+  //   });
+  // };
```
Comment (Shubhamkr2610) [inline src/components/ProductCard/index.jsx]: remove unused code

**Review by Shubhamkr2610 — APPROVED** [review]
your logic is good , code is clean . everything looks good to me . you can merge it

File: src/App.js [hunk 3]
```diff
@@ -10,24 +10,37 @@ import { AuthProvider } from "./Contexts/AuthDialogContext";
...
-              <Routes>
-                <Route path="/" exact element={<Home />} />
-                <Route path="/products" exact element={<ProductListing />} />
-                <Route path="/wishlist" exact element={<Wishlist />} />
-                <Route path="/cart" exact element={<Cart />} />
-                <Route path="/mock-api" element={<MockmanEs />} />
-              </Routes>
+              <WishlistProvider>
+                <CartProvider>
+                  {/* <Login /> */}
```
Comment (shivanigangadharan) [inline src/App.js:27]: can remove commented code

File: src/Contexts/CartContext/index.jsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    () =>
+      (async () => {
+        if (user) {
+          setIsCartLoading(true);
+          try {
+            const res = await axios.get("/api/user/cart", {
+              headers: { authorization: utils.getLocalStorage("authToken") },
+            });
+            if (res.status === 200) {
+							setCartProducts(res.data.cart);
```
Comment (shivanigangadharan) [inline src/Contexts/CartContext/index.jsx:27]: I like how you've handled errors here using status code.

File: src/Hooks/useProductCard.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    setAddingToCart(true);
+    try {
+      const res = await axios.delete(`/api/user/cart/${product._id}`, {
+        headers: { authorization: utils.getLocalStorage("authToken") },
+      });
+
+      if (res.status === 200) {
+        toast.success("removed from cart");
+        setAddingToCart(false);
+        // setCart((prev) => ({ ...prev, cartProducts: res.data.cart }));
```
Comment (shivanigangadharan) [inline src/Hooks/useProductCard.js:63]: can remove this

File: src/components/Authentication/Login.jsx [hunk 6]
```diff
@@ -75,10 +75,16 @@ const Login = () => {
...
             buttonText="Create new account"
-            buttonStyle="secondary-button"
+            buttonStyle="secondary-button margin-top-0"
             // icon="fas fa-chevron-right"
             onClick={() => setAuthType("signup")}
           />
+          <Button
+            buttonText="Login with test credenials"
+            buttonStyle="secondary-button body-typo-md margin-top-0"
+            // icon="fas fa-chevron-right"
```
Comment (shivanigangadharan) [inline src/components/Authentication/Login.jsx:85]: can remove this comment too

File: src/components/Header/Logo.jsx [hunk 7]
```diff
@@ -1,8 +1,10 @@
 import React from "react";
+import { Link } from "react-router-dom";
 const Logo = () => {
   return (
     <div className="nav-logo nav-section-items">
-      <a className="link-no-style" href="/">
+      {/* <Link to="/" className="link-no-style" > */}
+      {/* <a className="link-no-style" href="/"> */}
```
Comment (shivanigangadharan) [inline src/components/Header/Logo.jsx:7]: few more comments to remove

File: src/screens/Wishlist/index.jsx [hunk 8]
```diff
@@ -1,30 +1,40 @@
...
+              variant="horizontal"
+              cartActionBtnContainer={false}
+              product={product}
+              key={product._id}
+            />
+          );
+        })}
+      </section> : 
+      <div className="flex-column flex-align-item-center flex-justify-content-center margin-top-60">
+      <h2>Your cart is empty!</h2>
```
Comment (shivanigangadharan) [inline src/screens/Wishlist/index.jsx:35]: It should be "Your wishlist is empty" (instead of cart)

**Review by shivanigangadharan — APPROVED** [review]
Left a few comments, overall good code and nice UI as well. Can merge. 👍

File: src/screens/Wishlist/index.jsx (hunk 8 above)
Comment (abhishektanwar) [reply src/screens/Wishlist/index.jsx:35]: fixed

File: src/components/Header/Logo.jsx (hunk 7 above)
Comment (abhishektanwar) [reply src/components/Header/Logo.jsx:7]: removed

File: src/components/Authentication/Login.jsx (hunk 6 above)
Comment (abhishektanwar) [reply src/components/Authentication/Login.jsx:85]: removed

File: src/Hooks/useProductCard.js (hunk 5 above)
Comment (abhishektanwar) [reply src/Hooks/useProductCard.js:63]: removed

File: src/App.js (hunk 3 above)
Comment (abhishektanwar) [reply src/App.js:27]: removed

**Pull request merged**